lib/fresh_connection/connection_factory.rb

```diff
@@ -22,19 +25,66 @@ def spec
       @spec ||= build_spec
     end
 
+    # when building a spec from envars, there may be no database.yml file, so
+    # be careful to avoid implicit dependency on it or derived contents.
+
     def build_spec
-      config = ar_spec.config.symbolize_keys
-      group_config = config[@group]
+      url = group_url
+      if url
+        build_and_set_group_spec_and_url(url)
+      else
+        build_spec_from_config
+      end
+    end
+
+    # look for an envar with the current @group name, if it's not found, and if
+    # the group is :replica, then look for an envar using :replica1, which if
+    # found becomes the new @group
 
-      # provide backward compatibility for older :slave usage
-      if !group_config && @group == :replica && config.key?(:slave)
-        ActiveSupport::Deprecation.warn(
-          "'slave' in database.yml is deprecated and will ignored from version 2.4.0. use 'replica' insted."
-        )
-        group_config = config[:slave]
+    def group_url
+      url = database_group_url(@group.to_s)
+      if url.nil? && @group == :replica
+        url = database_group_url('replica1')
+        @group = :replica1 if url
       end
+      url
+    end
+
+    def database_group_url(group)
+      ENV['DATABASE_' + group.upcase + '_URL']
+    end
+
+    def build_and_set_group_spec_and_url(url)
+      config = ar_spec.config.with_indifferent_access
+      spec = build_spec_from_url(url)
+      group_config = (config[@group] ||= {}.with_indifferent_access)
+      config.merge(group_config).merge(spec).merge("url" => url).merge(@modify_spec)
```

Comment:
Since you can already get the setting from the url, you do not need to specify `"url"` again.
There is no problem with the following.

```ruby
config.merge(group_config).merge(spec).merge(@modify_spec)
```

`"url"` affects establish_connection, but it has already done the process of parsing the URL in establish_connection.
I think that it is not necessary to specify the URL anymore.